types/got/index.d.ts

```diff
@@ -98,15 +98,19 @@ declare namespace got {
         json?: boolean;
     }
 
-    type GotOptions<E extends string | null> = http.RequestOptions & {
+    interface TimoutRequestOptions extends http.RequestOptions {
+        timeout?: any;
```

Comment:
Instead of `any` this should be typed to valid input, something like: `number | TimeoutOptions`